src/backend/src/modules/puterai/AWSPollyService.js

```diff
@@ -64,25 +73,48 @@ class AWSPollyService extends BaseService {
             * @property {Object} synthesize - Converts text to speech using specified voice/language
             * @property {Function} supports_test_mode - Indicates test mode support for methods
             */
-            async list_voices () {
+            async list_voices ({ engine } = {}) {
                 const polly_voices = await this.describe_voices();
 
                 let voices = polly_voices.Voices;
 
+                // Filter voices by engine if specified
+                if (engine && VALID_ENGINES.includes(engine)) {
+                    voices = voices.filter((voice) => {
+                        // Check if voice supports the specified engine
+                        return voice.SupportedEngines && voice.SupportedEngines.includes(engine);
+                    });
+                }
+
                 voices = voices.map((voice) => ({
                     id: voice.Id,
                     name: voice.Name,
                     language: {
                         name: voice.LanguageName,
                         code: voice.LanguageCode,
                     },
+                    supported_engines: voice.SupportedEngines || ['standard'],
                 }))
 
                 return voices;
             },
+            async list_engines () {
+                return VALID_ENGINES.map(engine => ({
+                    id: engine,
+                    name: engine.charAt(0).toUpperCase() + engine.slice(1),
+                    pricing_per_million_chars: ENGINE_PRICING[engine] / 100, // Convert microcents to dollars
+                }));
+            },
+            async get_engine_voices (engine) {
+                if (!VALID_ENGINES.includes(engine)) {
+                    throw APIError.create('invalid_engine', { engine, valid_engines: VALID_ENGINES });
```

Comment:
The second parameter is reserved for a "source error" - an error object that represents the cause of the API error, or `null` if not applicable (usually `null`).

```suggestion
                    throw APIError.create('invalid_engine', null, { engine, valid_engines: VALID_ENGINES });
```

Additionally, `invalid_engine` should be registered with an error message in APIError.js